src/app/listing/listing.page.ts

```diff
@@ -17,37 +18,29 @@ export class ListingPage implements OnInit {
   loader: LoaderService;
   solutionId!: string;
   listType!: keyof UrlConfig;
-  route: ActivatedRoute;
   listTitle!: string;
   listDescription!: string;
   searchTerm: any = "";
-  toastService:ToastService;
+  toastService: ToastService;
+  stateData: any;
+  page:number = 1
 
-  constructor(private http: HttpClient) {
-    this.route = inject(ActivatedRoute);
+  constructor(private http: HttpClient, private navCtrl: NavController, private router: Router) {
     this.baseApiService = inject(ApiBaseService);
     this.loader = inject(LoaderService)
-     this.toastService = inject(ToastService)
+    this.toastService = inject(ToastService)
   }
 
   ngOnInit() {
-    this.route.params.subscribe((param: any) => {
-      console.log(param)
-      this.listType = param['type'];
-    });
-    this.http.get<any>('assets/listingData.json').subscribe(data => {
-      this.listData = data;
-      data.forEach((ele: any) => {
-        if (ele?.type == "solutionList") {
-          ele?.data?.forEach((res: any) => {
-            if (res?.listType == this.listType) {
-              this.listTitle = res?.name;
-              this.listDescription = res?.description;
-            }
-          });
-        }
-      });
-    });
+    const navigation = this.router.getCurrentNavigation();
+    if (navigation?.extras?.state) {
+      this.stateData = navigation.extras.state;
+      console.log('statee', this.stateData);
+      this.listTitle = this.stateData?.name;
```

Comment:
instead of assigning each item individually can we use object destructing to assign item at once? see if that can be done